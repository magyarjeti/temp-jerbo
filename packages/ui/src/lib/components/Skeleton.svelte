<script lang="ts">
	import { concatClasses } from '@packages/toolbox/concatClasses';
	let { ...props }: { class: string } = $props();
	const baseClasses = 'pulse bg-surface-container-highest opacity-1';
	const classes = $derived(concatClasses(props.class, baseClasses));
</script>

<div class={classes}><span class="sr-only">Betöltés alatt</span></div>

<style lang="postcss">
	.pulse {
		animation: 1s infinite ease-in pulse;
	}

	@keyframes pulse {
		0% {
			opacity: 1;
		}

		50% {
			opacity: 0.5;
		}

		100% {
			opacity: 1;
		}
	}
</style>
