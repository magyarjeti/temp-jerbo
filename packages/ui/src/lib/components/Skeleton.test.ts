import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import Skeleton from './Skeleton.svelte';

test('Skeleton snapshot', () => {
	const { container } = render(Skeleton);
	expect(container).toMatchSnapshot();
});
