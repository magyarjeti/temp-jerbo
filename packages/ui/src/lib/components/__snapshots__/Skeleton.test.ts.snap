// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton snapshot 1`] = `
<div>
  <div
    class="pulse bg-surface-container-highest opacity-1 svelte-54far3"
  >
    <span
      class="sr-only"
    >
      Betöltés alatt
    </span>
  </div>
</div>
`;
